staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -307,7 +307,10 @@ func (e *eventBroadcasterImpl) StartStructuredLogging(verbosity klog.Level) func
 // StartEventWatcher starts sending events received from this EventBroadcaster to the given event handler function.
 // The return value is used to stop recording
 func (e *eventBroadcasterImpl) StartEventWatcher(eventHandler func(event runtime.Object)) func() {
-	watcher := e.Watch()
+	watcher, err := e.Watch()
+	if err != nil {
+		klog.Errorf("Unable start event watcher: '%v' (will not retry!)", err)
```

Comment:
This has a potential to cause - if e.Watch() returns an error, it may return nil, then calling anything on that isn't safe.

We should return some noop func here probably.

Reply:
Yeah I see.. I would rather stop the whole flow from continuing but for now I'll just add no-op function that continuously logs the error

Reply:
Added a TODO comment to fix this further in a follow-up PR